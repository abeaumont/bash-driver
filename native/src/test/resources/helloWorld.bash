#!/usr/bin/env bash

function say {
    echo $1
}

say "hello world"
