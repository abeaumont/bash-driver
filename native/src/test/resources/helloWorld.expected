{"status":"ok","errors":[],"ast":{"elementType":"FILE","startOffset":0,"textLength":69,"text":"#!/usr/bin/env bash\n\nfunction say {\n    echo $1\n}\n\nsay \"hello world\"\n","children":[{"elementType":"[Bash] shebang element","startOffset":0,"textLength":20,"text":"#!/usr/bin/env bash\n","children":[{"elementType":"[Bash] Shebang","startOffset":0,"textLength":20,"text":"#!/usr/bin/env bash\n","children":[]}]},{"elementType":"[Bash] linefeed","startOffset":20,"textLength":1,"text":"\n","children":[]},{"elementType":"function-def-element","startOffset":21,"textLength":28,"text":"function say {\n    echo $1\n}","children":[{"elementType":"[Bash] function","startOffset":21,"textLength":8,"text":"function","children":[]},{"elementType":"WHITE_SPACE","startOffset":29,"textLength":1,"text":" ","children":[]},{"elementType":"[Bash] named symbol","startOffset":30,"textLength":3,"text":"say","children":[{"elementType":"[Bash] word","startOffset":30,"textLength":3,"text":"say","children":[]}]},{"elementType":"WHITE_SPACE","startOffset":33,"textLength":1,"text":" ","children":[]},{"elementType":"group element","startOffset":34,"textLength":15,"text":"{\n    echo $1\n}","children":[{"elementType":"[Bash] {","startOffset":34,"textLength":1,"text":"{","children":[]},{"elementType":"[Bash] linefeed","startOffset":35,"textLength":1,"text":"\n","children":[]},{"elementType":"WHITE_SPACE","startOffset":36,"textLength":1,"text":" ","children":[]},{"elementType":"WHITE_SPACE","startOffset":37,"textLength":1,"text":" ","children":[]},{"elementType":"WHITE_SPACE","startOffset":38,"textLength":1,"text":" ","children":[]},{"elementType":"WHITE_SPACE","startOffset":39,"textLength":1,"text":" ","children":[]},{"elementType":"simple-command","startOffset":40,"textLength":7,"text":"echo $1","children":[{"elementType":"[Bash] generic bash command","startOffset":40,"textLength":4,"text":"echo","children":[{"elementType":"[Bash] combined word","startOffset":40,"textLength":4,"text":"echo","children":[{"elementType":"[Bash] word","startOffset":40,"textLength":4,"text":"echo","children":[]}]}]},{"elementType":"WHITE_SPACE","startOffset":44,"textLength":1,"text":" ","children":[]},{"elementType":"[Bash] combined word","startOffset":45,"textLength":2,"text":"$1","children":[{"elementType":"var-use-element","startOffset":45,"textLength":2,"text":"$1","children":[{"elementType":"[Bash] variable","startOffset":45,"textLength":2,"text":"$1","children":[]}]}]}]},{"elementType":"[Bash] linefeed","startOffset":47,"textLength":1,"text":"\n","children":[]},{"elementType":"[Bash] }","startOffset":48,"textLength":1,"text":"}","children":[]}]}]},{"elementType":"[Bash] linefeed","startOffset":49,"textLength":1,"text":"\n","children":[]},{"elementType":"[Bash] linefeed","startOffset":50,"textLength":1,"text":"\n","children":[]},{"elementType":"simple-command","startOffset":51,"textLength":17,"text":"say \"hello world\"","children":[{"elementType":"[Bash] generic bash command","startOffset":51,"textLength":3,"text":"say","children":[{"elementType":"[Bash] combined word","startOffset":51,"textLength":3,"text":"say","children":[{"elementType":"[Bash] word","startOffset":51,"textLength":3,"text":"say","children":[]}]}]},{"elementType":"WHITE_SPACE","startOffset":54,"textLength":1,"text":" ","children":[]},{"elementType":"[Bash] string","startOffset":55,"textLength":13,"text":"\"hello world\"","children":[{"elementType":"[Bash] string begin","startOffset":55,"textLength":1,"text":"\"","children":[]},{"elementType":"[Bash] string content","startOffset":56,"textLength":11,"text":"hello world","children":[]},{"elementType":"[Bash] string end","startOffset":67,"textLength":1,"text":"\"","children":[]}]}]},{"elementType":"[Bash] linefeed","startOffset":68,"textLength":1,"text":"\n","children":[]}]}}
